File: app/src/main/java/com/metahumanz/pacilread/model/ReaderThemeRecord.java
package com.metahumanz.pacilread.model;

import org.json.JSONObject;

public class ReaderThemeRecord {
    public long id;
    public String name;
    public String configJson;
    public long updatedAt;

    public JSONObject toJson() {
        JSONObject json = new JSONObject();
        try {
            json.put("id", id);
            json.put("name", name != null ? name : "");
            json.put("configJson", configJson != null ? configJson : "{}");
            json.put("updatedAt", updatedAt);
        } catch (Exception ignore) {}
        return json;
    }

    public static ReaderThemeRecord fromJson(JSONObject json) {
        ReaderThemeRecord theme = new ReaderThemeRecord();
        theme.id = json.optLong("id", 0);
        theme.name = json.optString("name", "");
        theme.configJson = json.optString("configJson", "{}");
        theme.updatedAt = json.optLong("updatedAt", 0);
        return theme;
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/model/ReaderThemeRecord.kt
package com.metahumanz.pacilread.model

import org.json.JSONObject

class ReaderThemeRecord {
    @JvmField var id: Long = 0
    @JvmField var name: String? = null
    @JvmField var configJson: String? = null
    @JvmField var updatedAt: Long = 0

    fun toJson(): JSONObject {
        val json = JSONObject()
        try {
            json.put("id", id)
            json.put("name", name ?: "")
            json.put("configJson", configJson ?: "{}")
            json.put("updatedAt", updatedAt)
        } catch (_: Exception) {
        }
        return json
    }

    companion object {
        @JvmStatic
        fun fromJson(json: JSONObject): ReaderThemeRecord = ReaderThemeRecord().apply {
            id = json.optLong("id", 0)
            name = json.optString("name", "")
            configJson = json.optString("configJson", "{}")
            updatedAt = json.optLong("updatedAt", 0)
        }
    }
}
